autorest/responder.go

```diff
@@ -142,23 +142,34 @@ func ByUnmarshallingJSON(v interface{}) RespondDecorator {
 	}
 }
 
-// ByUnmarshallingSuccessandErrorJSON returns a RespondDecorator that decodes a JSON document returned in the
+// WithServiceErrorUnlessOK returns a RespondDecorator that decodes a JSON document returned in the
 // response Body into the value pointed by success schema v or error schema e
-func ByUnmarshallingSuccessandErrorJSON(v interface{}, e interface{}) RespondDecorator {
+func WithServiceErrorUnlessOK(v interface{}, e interface{}) RespondDecorator {
 	return func(r Responder) Responder {
```

Comment:
i tried to be consistent. I didn't liked v either. I will make it more readable.